NuRadioReco/utilities/geometryUtilities.py

```diff
@@ -162,10 +236,88 @@ def get_fresnel_r_s(zenith_incoming, n_2=1.3, n_1=1.):
     this polarization corresponds to the ePhi polarization
 
     parallel and perpendicular refers to the signal's polarization with respect
-    to the 'plane of incident' which is defindes as: "the plane of incidence
+    to the 'plane of incident' which is defined as: "the plane of incidence
     is the plane which contains the surface normal and the propagation vector
     of the incoming radiation."
+
+    Parameters
+    ----------
+    zenith_incoming: float
+        Zenith angle of the incoming signal
+    n_2: float
+        Refractive index of the medium the signal is reflected from
+    n_1: float
+        Refractive index of the medium the signal is coming from
+
+    Returns
+    -------
+    float
+        Fresnel coefficient r for phi (perpendicular) polarization
     """
     n = n_2 / n_1
     return np.conjugate((np.cos(zenith_incoming) - SM.sqrt(n**2 - np.sin(zenith_incoming)**2)) / \
                         (np.cos(zenith_incoming) + SM.sqrt(n**2 - np.sin(zenith_incoming)**2)))
+
+
+def fresnel_factors_and_signal_zenith(detector, station, channel_id, zenith):
+    """
+    Returns the zenith angle at the antenna and the fresnel coefficients t for theta (parallel) 
+    and phi (perpendicular) polarization. Handles potential refraction into the firn if that 
+    applies to the antenna position. 
+    WARNING: for deeper channels this function might be inacccurate. Consider using raytracing.
+
+    parallel and perpendicular refers to the signal's polarization with respect
+    to the 'plane of incident' which is defined as: "the plane of incidence
+    is the plane which contains the surface normal and the propagation vector
+    of the incoming radiation.
+
+    Parameters
+    ----------
+    detector: NuRadioReco.detector.Detector
+        Detector object
+    station: NuRadioReco.detector.Station
+        Station object
+    channel_id: int
+        Channel ID of the desired channel
+    zenith: float
+        Zenith angle of the incoming signal
+    
+    Returns
+    -------
+    zenith_antenna: float
+        Zenith angle at the antenna (potentially including refraction)
+    t_theta: float
+        Fresnel transmission coefficient for theta polarization
+    t_phi: float
+        Fresnel transmission coefficient for phi polarization
+    """
+    n_ice = ice.get_refractive_index(-0.01, detector.get_site(station.get_id()))
+
+    # no reflection/refraction at the ice-air boundary
+    zenith_antenna = zenith
+    t_theta = 1.
+    t_phi = 1.
+
+    position = detector.get_relative_position(station.get_id(), channel_id)
+    if position[2] < -3 * units.m:
+        logger.warning("This function might return inaccurate results. Consider using raytracing instead")
```

Comment:
Thanks for adding a warning! Can you make it slightly more specific and say `This function might return inaccurate results for deep in-ice antennas`?